src/architecture/storage.md

```diff
@@ -1,262 +1,158 @@
 
-Currently the main database engine is [RocksDB][rocks-db].
+- `transactions: MapIndex`  
+  Represents a map from the transaction hash into a raw transaction structure.
+- `transaction_results: ProofMapIndex`  
+  Keeps execution results for all accepted transactions,
+  indexed by transaction hashes.
+- `transactions_pool: KeySetIndex`  
+  Stores the set of hashes of the known transactions that have not been
+  committed yet.
+- `transactions_pool_len: Entry`  
+  Caches the number of entries in `transactions_pool`.
+- `transactions_locations: MapIndex`  
+  For every transaction hash keeps the position of said transaction inside the
+  block and the block height.
+- `blocks: MapIndex`  
+  Stores the block object for every block height.
+- `block_hashes_by_height: ListIndex`  
+  Saves block hashes indexed by block heights.
+- `block_transactions: ProofListIndex`  
+  Stores a list of transactions of the specific block indexed by the block
+  height.
+- `precommits: ListIndex`  
+  Stores a list of precommits of the validators for the specific block, indexed
+  by the block hash.
+- `configs: ProofMapIndex`  
+  Stores the configuration content in JSON format using its hash as a key.
+- `configs_actual_from: ListIndex`  
+  Stores the hashes of the upcoming configurations indexed by the heights, where
+  each configuration should become active. The list allows a simple search for
+  the upcoming configuration hash by its height. The discovered hash allows a
+  simple search of the corresponding configuration in the
+  `configs: ProofMapIndex`.
+- `state_hash_aggregator: ProofMapIndex`  
+  An accessory store for summing up the state hash of the whole blockchain.
 
-Values from different tables are stored in column families in the low-level
-storage,
-wherein the keys are represented as
-a byte sequence, and values are serialized according to Protobuf
-serialization format. A single column family may store data for
-more than one table (see table groups below).
-Keys of a specific table are mapped to the low-level storage keys
-in a deterministic manner using [table identifiers](#table-identifiers).
+## Low-level Storage
 
-### Table Identifiers
+Exonum uses a third-party database engine to persist blockchain state
+locally. Currently the main database engine is [RocksDB][rocks-db]. It is also
+possible to plug in other engines.
 
-Every table is uniquely identified by a compound identifier, which is used
-to map table keys into a column family and its keys in the underlying
-low-level storage. A table identifier consists of 2 parts:
+To use a particular database, a minimal [`Database`][database]
+interface should be implemented for it:
 
-- **String name,** which is mapped 1-to-1 to a column family.
-  The name may contain uppercase and lowercase Latin letters, digits,
-  underscores `_`, and periods `.`. By convention, table names in services
+- Get a value by a [column family][col-family] name and a key
```

Comment:
Is it appropriate to link Rocksdb-specific article when talking about _our_ interface for plugging any database engines?